packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -376,9 +454,29 @@ class BabelPluginGetPluginsFromGatsbyConfig {
               return
             }
 
+            // Only allow module.exports = {}
+            // Not things like module.exports = () => ({})
+            if (!t.isObjectExpression(right)) {
+              return
+            }
+
             const plugins = right.properties.find(p => p.key.name === `plugins`)
 
-            plugins.value.elements.map(node => {
+            let pluginsList = []
+
+            if (t.isCallExpression(plugins.value)) {
+              pluginsList = plugins?.value?.callee?.object?.elements
+            } else {
+              pluginsList = plugins?.value?.elements
```

Comment:
If it didn't crash before it won't crash now
```suggestion
              pluginsList = plugins.value.elements
```